File: app/src/main/java/com/marlon/portalusuario/ViewModel/UneViewModel.java
package com.marlon.portalusuario.ViewModel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.marlon.portalusuario.PUNotifications.PUNotification;
import com.marlon.portalusuario.User;
import com.marlon.portalusuario.database.notifications.PunRepository;
import com.marlon.portalusuario.database.une.UneRepository;
import com.marlon.portalusuario.database.users.UserRepository;
import com.marlon.portalusuario.une.Une;

import java.util.List;

public class UneViewModel extends AndroidViewModel {
    private UneRepository uneRepository;

    public UneViewModel(@NonNull Application application) {
        super(application);
        uneRepository = new UneRepository(application);
    }

    // UNE
    public void insertUne(Une une){
        uneRepository.insertUne(une);
    }

    public void updateUne(Une une){
        uneRepository.updateUne(une);
    }

    public void deleteUne(Une une){
        uneRepository.deleteUne(une);
    }

    public void deleteAllUnes(){
        uneRepository.deleteAllUnes();
    }

    public LiveData<List<Une>> getAllUnes(){
        return uneRepository.getAllUne();
    }
}

File: app/src/main/java/com/marlon/portalusuario/viewmodel/UneViewModel.kt
package com.marlon.portalusuario.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.asLiveData
import androidx.lifecycle.viewModelScope
import com.marlon.portalusuario.database.une.UneRepository
import com.marlon.portalusuario.une.Une
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.SharingStarted
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.stateIn
import kotlinx.coroutines.launch
import javax.inject.Inject

@HiltViewModel
class UneViewModel
    @Inject
    constructor(
        application: Application,
    ) : AndroidViewModel(application) {
        private val uneRepository = UneRepository(application)

        val allUnes: StateFlow<List<Une>> =
            uneRepository.allUnes
                .stateIn(
                    scope = viewModelScope,
                    started = SharingStarted.WhileSubscribed(5_000),
                    initialValue = emptyList(),
                )

        val allUnesLiveData: LiveData<List<Une>> = allUnes.asLiveData()

        fun insertUne(une: Une) {
            viewModelScope.launch {
                uneRepository.insertUne(une)
            }
        }

        fun updateUne(une: Une) {
            viewModelScope.launch {
                uneRepository.updateUne(une)
            }
        }

        fun deleteUne(une: Une) {
            viewModelScope.launch {
                uneRepository.deleteUne(une)
            }
        }

        fun deleteAllUnes() {
            viewModelScope.launch {
                uneRepository.deleteAllUnes()
            }
        }
    }
